### PR TITLE
(feat-multi-cluster):block resources based on kuma-cp mode

### LINKED ISSUE #767: Block Dataplane API/ XDS/SDS in Global
### Summary

When Kuma CP is run in a Global mode, disable:
 * Block the Dataplane REST API
 * XDS server for the Envoy configuration
 * SDS server for distributing the secrets

### PR DESCRIPTION
**Pull request by tharun208**
### Summary

Updated `kuma-cp` run to block resources based on the mode.

### Full changelog

* Updated kuma-cp run

### Issues resolved

Fix #767 #768

### CODE REVIEW

File: app/kuma-cp/cmd/run.go [hunk 1]
```diff
@@ -71,29 +71,33 @@ func newRunCmdWithOpts(opts runCmdOpts) *cobra.Command {
...
+				if err := xds_server.SetupServer(rt); err != nil {
+					runLog.Error(err, "unable to set up xDS server")
+					return err
+				}
+				if err := api_server.SetupServer(rt); err != nil {
+					runLog.Error(err, "unable to set up API server")
+					return err
+				}
 			}
 			if err := mads_server.SetupServer(rt); err != nil {
```
Comment (jakubdyszkiewicz) [inline app/kuma-cp/cmd/run.go:88]: MADS should also be run only in local and standalone.

File: pkg/api-server/server.go [hunk 2]
```diff
@@ -164,7 +165,11 @@ func (a *ApiServer) Start(stop <-chan struct{}) error {
 
 func SetupServer(rt runtime.Runtime) error {
 	cfg := rt.Config()
-	apiServer, err := NewApiServer(rt.ResourceManager(), definitions.All, rt.Config().ApiServer, &cfg)
+	var apiDefinitions = definitions.All
+	if cfg.Mode == config_core.Local {
+		apiDefinitions = definitions.DataPlaneDefinitions
+	}
```
Comment (jakubdyszkiewicz) [inline pkg/api-server/server.go:171]: I think for local it makes sense to expose other APIs that Dataplane but in read-only mode. We want to have ability to debug if resources are synchronized.

@nickolaev WDYT?

**Comment by nickolaev** [comment]
OK, there are actually some problems here. First, we have failing e2e tests. Can you verify these works locally for you?

Then:
 * Local in K8s we want the API to be ReadOnly. It is such by default, so no issues here.
 * When you run Local in Universal, we want the API to allow Updates to Dataplaens resources only.

File: pkg/api-server/server.go (hunk 2 above)
Comment (jakubdyszkiewicz) [reply pkg/api-server/server.go:170]: @tharun208 read only is for other resources than dataplane.

**Comment by nickolaev** [comment]
The failure in the e2e test is actually verification that API is working or not.
In this case it fails for global:
https://github.com/Kong/kuma/blob/master/test/e2e/kuma_deploy_global_local_test.go#L22

Can you please check that when you deploy by hand you get Global API and Local API blocked.
You will also have to change:
https://github.com/Kong/kuma/blob/master/test/e2e/kuma_deploy_global_local_test.go#L25
to `Expect(err).To(HaveOccurred())` cause we expect in Local to not have access to the API.

**Comment by nickolaev** [comment]
@tharun208 the e2e tests in master a much more stable now, please merge it and reconsider this PR on top of it.

**Comment by nickolaev** [comment]
OK, the failing tests is fine (we can not register kumactl as there is no API). I will merge this one and fix it as a follow up.

**Pull request merged**